src/charm.py

```diff
@@ -232,14 +233,21 @@ def _reconcile(self) -> None:
                     self._ingress_provider
                 )
                 tls_information = TLSInformation.from_charm(self, self.certificates)
+                self.unit.set_ports(80, 443)
                 self.haproxy_service.reconcile_ingress(
                     config, ingress_requirers_information, tls_information.external_hostname
                 )
             case ProxyMode.LEGACY:
+                required_ports = set(
+                    Port(protocol="tcp", port=service["service_port"])
+                    for service in self.reverseproxy_requirer.get_services_definition().values()
```

Comment:
It would be comforting to add a type for the set (`set[int]`) and/or for the return value of `self.reverseproxy_requirer.get_services_definition()` to make sure that we're handling the right data. Is that possible ?

Reply:
hm, sorry, I can't see it. Where did you add it ?

Reply:
oops, pushed it to the wrong PR, sorry. I just correctly updated it